carcara/src/lib.rs

```diff
@@ -130,38 +135,134 @@ pub enum Error {
     DoesNotReachEmptyClause,
 }
 
-pub fn check<T: io::BufRead>(problem: T, proof: T, options: CarcaraOptions) -> Result<bool, Error> {
-    let (prelude, proof, mut pool) = parser::parse_instance(
+pub fn check<T: io::BufRead>(
```

Comment:
I think it would be better if we still allowed the user to easily check using the single-threaded checker. Ideally, I think there should be two check functions, `check`, that uses the single-threaded checker and has the same behaviour it had before, and `check_parallel`, like you implemented here, which receives a number of threads and stack size. Then in the CLI, if the `--num-threads` option is 1, we use `check`, otherwise we use `check_parallel`